Content.Server/Procedural/DungeonSystem.Rooms.cs

```diff
@@ -118,7 +132,12 @@ public void SpawnRoom(
         // go BRRNNTTT on existing stuff
         if (clearExisting)
         {
-            var gridBounds = new Box2(Vector2.Transform(-room.Size/2, roomTransform), Vector2.Transform(room.Size/2, roomTransform));
+            //The Box2 rotation completely breaks the entity calculation from lookup. and before that, there's a 75% chance the spawn room won't remove anything underneath it.
+            //Therefore, Box2 must be calculated separately for all 4 rotation options.
+            var point1 = Vector2.Transform(-room.Size / 2, roomTransform);
+            var point2 = Vector2.Transform(room.Size / 2, roomTransform);
+            var gridBounds = GetRotatedBox(point1, point2, finalRoomRotation);
+
```

Comment:
Is the rounding on it that bad? I would think for a lookup in particular it's close enough? Worst case I imagine you would just .Floor / .Ceil it? I don't think GetRotatedBox is necessary.